haematite_ser/src/wrap.rs

```diff
@@ -0,0 +1,242 @@
+use std::collections::HashMap;
+
+use haematite_models::meta::permissions::Tree;
+use serde::ser::{Serialize, Serializer};
+use serde::ser::{
+    SerializeMap, SerializeSeq, SerializeStruct, SerializeStructVariant, SerializeTuple,
+    SerializeTupleStruct, SerializeTupleVariant,
+};
+
+#[derive(Debug, Clone)]
+pub struct SerializeWrap {
+    pub inner: WrapType,
+    pub allowed: bool,
+}
+
+impl SerializeWrap {
+    pub fn new(inner: WrapType) -> Self {
+        Self {
+            inner,
+            allowed: true,
```

Comment:
Perhaps fail closed instead (untested)? It is slightly uncomfortable that the `match` in `update_with` below has a default case that does nothing.
```suggestion
            allowed: false,
```

Reply:
the reason it is `true` because default is there's a line directly after it gets wrapped (in `haematite_bin`) that prints the whole thing to terminal for debugging. this will be removed later, or at least gated behind a cli arg.

instead of making this fail-closed, i intended to thoroughly test `update_with`